modules/ROOT/pages/dataweave-extension-plugin.adoc

```diff
@@ -0,0 +1,362 @@
+
+* Click the *DataWeave: Run Preview* button (arrow icon) at the top right corner of your mapping `.dwl` file window.
++
+image::dataweave-extension-preview-1.png[DataWeave: Run Preview button]
+
+* Right-click the editor and select *DataWeave: Enable AutoPreview* to run it on every file change.
++
+image::dataweave-extension-preview-2.png[DataWeave: Enable AutoPreview option]
+
+
+
+== Run and Debug a DataWeave Mapping
+
+Run and debug a DataWeave mapping to be guided through all of the mapping elements, which you can then fix or otherwise update. You can also set breakpoints to stop evaluating expressions and use all the debugging capabilities that Visual Studio Code provides. Debugging works on mapping tests as well.
+
+To run and debug a DataWeave mapping, follow these steps:
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/main/dw` folder.
+. Select your mapping `.dwl` file.
+. Click the *Run Mapping* button located at the top of your `.dwl` file.
++
+image::dataweave-extension-runmapping.png[MyMapping.dwl and Run Mapping button]
+
+
+== Test a DataWeave Mapping
+
+To test a DataWeave mapping, click the *Dataweave: Create Mapping Test* button located at the top of your `.dwl` file to create a new test scenario from a snapshot of your input and your current output.
+You can also use this newly created scenario as a normal scenario for your preview or autocompletion.
+
+To test a DataWeave mapping, follow these steps:
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/main/dw` folder.
+. Click the *DataWeave: Create Mapping Test* button (microscope icon) located at the top of your mapping `.dwl` file.
++
+image::dataweave-extension-testmapping.png[MyMapping.dwl file and the DataWeave Create Mapping Test button]
+[start=4]
+. Specify a name for the mapping test.
+
+
+== Using the DataWeave Dependencies View
+
+The *DATAWEAVE DEPENDENCIES* view shows all the dependencies resolved for a DataWeave project. The view enables you to navigate through all of your project's dependencies and open DataWeave files inside those dependencies.
+
+image::dataweave-extension-dataweavedependencies.png[DataWeave Dependencies View]
+
+
+== Deploy DataWeave Libraries
+
+Before deploying your DataWeave library to Anypoint Exchange, generate documentation for your mapping or module functions. When you deploy your DataWeave library to Exchange, the documentation is also published to the Exchange library page.
+
+To generate documentation for the functions, follow these steps:
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/main/dw` folder.
+. Click the *Generate Weave Documentation* button located at the top of the function in the `.dwl` file.
++
+image::dataweave-extension-generatedocs-1.png[MyMapping.dwl file and the Generate Weave Documentation button]
```

Comment:
This should be a module. Not sure if docs for functions inside mappings are rendered in exchange because those functions are not exposed.